RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -97,16 +97,49 @@ class ManageSubscriptionsViewModel: ObservableObject {
     private func loadSubscriptionInformation() async throws {
         let customerInfo = try await purchasesProvider.customerInfo()
 
-        guard let currentEntitlement = customerInfo.earliestExpiringAppStoreEntitlement(),
-              let subscribedProduct = await purchasesProvider.products([currentEntitlement.productIdentifier]).first
+        // Find earliest expiring subscription
+        guard let earliestProductId =
```

Comment:
This code is a bit crazy, but I am planning on refactoring it by moving all the logic earlier in the flow. It would be helpful to get the Subscription information in the `CustomerInfo` so until then, we can leave it like this